src/main/java/picard/sam/AddOATag.java

```diff
@@ -0,0 +1,105 @@
+            "that fall within the intervals in the interval list" +
+            "<br />"+
+            "<h4>Usage example:</h4>" +
+            "<pre>" +
+            "java -jar picard.jar AddOATag \\<br />" +
+            "      L=some_picard.interval_list \\<br />" +
+            "      I=sorted.bam \\<br />" +
+            "      O=fixed.bam <br />"+
+            "</pre>";
+
+    @Argument(shortName = StandardOptionDefinitions.INPUT_SHORT_NAME, doc = "SAM or BAM input file")
+    public File INPUT;
+
+    @Argument(shortName = StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc = "SAM or BAM file to write merged result to")
+    public File OUTPUT;
+
+    @Argument(shortName = "OV", doc = "Whether or not to override any existing OA tag values", optional = true)
+    public Boolean OVERWRITE_TAG = true;
+
+    @Argument(shortName = "L", doc = "An interval list file that contains the locations of reads to add the OA tag to", optional = true)
+    public File INTERVAL_LIST;
+
+    private static final Log log = Log.getInstance(AddOATag.class);
+
+    public static void main(final String[] argv) {
+        System.exit(new AddOATag().instanceMain(argv));
+    }
+
+    @Override
+    protected int doWork() {
+        try (final SamReader reader = SamReaderFactory.makeDefault().referenceSequence(REFERENCE_SEQUENCE).open(INPUT)) {
+            try (final SAMFileWriter writer = new SAMFileWriterFactory().makeSAMOrBAMWriter(reader.getFileHeader(), true, OUTPUT)) {
+                writer.setProgressLogger(
+                        new ProgressLogger(log, (int) 1e7, "Wrote", "records"));
+
+                OverlapDetector overlapDetector = getOverlapDetectorFromIntervalListFile(INTERVAL_LIST, 0, 0);
+                for (SAMRecord rec : reader) {
+                    if (!rec.getReadUnmappedFlag() && (overlapDetector == null || overlapDetector.overlapsAny(rec))) {
+                        setOATag(rec, OVERWRITE_TAG);
+                    }
+                    writer.addAlignment(rec);
+                }
+            }
+        } catch (IOException e) {
+            log.error("Exception running AddOATag tool: " + e.getMessage());
+            return 1;
+        }
+
+        return 0;
+    }
+
+    // Take an interval list file and convert it to an overlap detector, can add left and right padding
+    static OverlapDetector<Interval> getOverlapDetectorFromIntervalListFile(File intervalList, int lhsBuffer, int rhsBuffer) {
+        if (intervalList == null) {
+            return null;
+        } else {
+            List<Interval> intervals = IntervalList.fromFile(intervalList).uniqued().getIntervals();
+            OverlapDetector<Interval> detector = new OverlapDetector<>(lhsBuffer, rhsBuffer);
+            detector.addAll(intervals, intervals);
+            return detector;
+        }
+    }
+
+    // format OA tag string according to the spec
+    private void setOATag(SAMRecord rec, Boolean overrideTag) {
```

Comment:
Would be nice to add a TODO to backport to htsjdk in an utility class for `SAMRecord`